src/main/scala/elo/Bot.scala

```diff
@@ -124,28 +120,31 @@ object PingPongBot extends App {
         slackClient.sendMessage(message.channel, leaderboard)
       }
 
-      // get ID of first other user mentioned
-      val challengee = mentionedIds
-        .find(_ != selfId) // retrieves first element for which the find condition is true
-        .map(PlayerService.playerService.getPlayer)
-        .getOrElse(Player.Nobody)
+      // this will never throw an exception since messages necessarily have a user
+      val challengerOpt = PlayerService.playerService.getPlayer(message.user)
+
+      // get first other player mentioned
+      val challengeeID = mentionedIds // seq[string]
+        .find(_ != selfId) // option[string]
+        .getOrElse("nobody")
+      val challengeeOpt = PlayerService.playerService.getPlayer(challengeeID)
 
       // if its a challenge, send a challenge message
       if(message.text.contains("hallenge")) {
-        val challenger = PlayerService.playerService.getPlayer(message.user)
-        val chalMessage = challengeMessage(challenger, challengee)
+        // need valid identities for both players for a challenge
+        val chalMessage = (challengerOpt, challengeeOpt) match {
```

Comment:
@wongkk25 what about this...having both challenger and challengee as options and pattern matching that we have both of them in order to generate a challenge / report a loss

Reply:
Oops just saw this comment, my response is here https://github.com/bticknor/elo-ranking-slackbot/pull/10#discussion_r257334864